Oct 3112 broadcast configuration tool missing documentation (#16995)
nrfconnect/sdk-nrf: samples/bluetooth/broadcast_config_tool/README.rst

.. _broadcast_configuration_tool:

Broadcast Configuration Tool
############################

.. contents::
   :local:
   :depth: 2

The Broadcast Configuration Tool application implements the :ref:`BIS gateway mode <nrf53_audio_app_overview>`.

> In this mode, transmitting broadcast audio happens using Broadcast Isochronous Stream (BIS) and Broadcast Isochronous Group (BIG).

Reviewer: Should mention Auracast here as well.

Reviewer: Nit 2.0. Depending on the config, it may or may not be an Auracaster.
Suggest:

The Broadcast Configuration Tool sample implements the :ref:`BIS gateway mode <nrf53_audio_app_overview>` and may act as an Auracast broadcaster if an Auracast preset is used.

The tool uses a shell interface to be able to configure the broadcast source in many different ways.

The following limitations apply to this sample application:

* Audio input: Pre-encoded LC3 data from an SD-card.

> * Maximum two BIG with four BIS streams each.

Reviewer: The first time BIG and BIS is mentioned, we should write it out in full.

Author: It is written out on line:12

.. _broadcast_configuration_tool_requirements:

Requirements
************

.. contents::
   :local:
   :depth: 2

The Broadcast Configuration Tool application is designed to be used only with the following hardware:

.. table-from-rows:: /includes/sample_board_rows.txt
   :header: heading
   :rows: nrf5340_audio_dk_nrf5340

.. note::
   The application supports PCA10121 revisions 1.0.0 or above.
   The application is also compatible with the following pre-launch revisions:

   * Revisions 0.8.0 and above.

.. _broadcast_configuration_tool_ui:

Shell commands list
**************

The Broadcast Configuration Tool uses a shell interface for all user interactions. The command used is

  .. parsed-literal::
     :class: highlight

     bct

> When describing the commands ``<>`` means that the arguments is mandatory and ``[]`` means that the argument is optional.

Reviewer: Is this the standard notation?

Author: I saw this being used in another documentation in NCS, but I don't know if that is the standard

                        RTN: 2
                        SDU size: 60
                        Max Transport Latency: 10 ms
                        Frame Interval: 10000 us
                        Presentation Delay: 40000 us
                Sampling rate: 24000 Hz
                Bitrate: 48000 bps
                Frame duration: 10000 us
                Octets per frame: 60
                Language: eng
                Context(s):
                        Live
                Program info: Mathematics 101
                Immediate rendering flag: set
                Number of BIS: 1
                Location:
                        BIS 0: Mono Audio

packing - Set type of packing
=============================

The command sets the type of packing for the BIS streams, this can be either sequential or interleaved.

Usage:
  .. parsed-literal::
     :class: highlight

     bct packing <seq/int> <BIG index>

For example:
  .. parsed-literal::
     :class: highlight

     bct packing int 0

This will set the packing for BIG0 to interleaved.

preset - Set preset
===================

Usage:
  .. parsed-literal::
     :class: highlight

     bct preset <preset> <BIG index> [subgroup index]

> The command sets the preset for a BIG or subgroup.

Reviewer: Suggest we elaborate a bit on what presets are in this context.

The tool uses a shell interface to be able to configure the broadcast source in many different ways.

The following limitations apply to this sample application:

* Audio input: Pre-encoded LC3 data from an SD-card.
* Maximum two BIG with four BIS streams each.

.. _broadcast_configuration_tool_requirements:

Requirements
************

.. contents::
   :local:
   :depth: 2

The Broadcast Configuration Tool application is designed to be used only with the following hardware:

.. table-from-rows:: /includes/sample_board_rows.txt
   :header: heading
   :rows: nrf5340_audio_dk_nrf5340

.. note::
   The application supports PCA10121 revisions 1.0.0 or above.
   The application is also compatible with the following pre-launch revisions:

   * Revisions 0.8.0 and above.

.. _broadcast_configuration_tool_ui:

Shell commands list
*******************

The Broadcast Configuration Tool uses a shell interface for all user interactions. The command used is

  .. parsed-literal::
     :class: highlight

     bct

When describing the commands ``<>`` means that the arguments is mandatory and ``[]`` means that the argument is optional.

The following shell commands are available:

> start - Start broadcaster

Author: It is also mentioned a bit here: http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-16995/nrf/samples/bluetooth/broadcast_config_tool/README.html#id36 but I can make the link between `bct show` and what is on air, meaning: If you don't see anything using bct show you won't see anything on air

> start - Start broadcaster

Reviewer 2: That makes it clear - nice!

> The command starts the broadcaster with the current configuration. If no configuration is set, the broadcaster will not start. To view the current configuration, use the ``show`` command.

Reviewer 3: ```suggestion
The command starts the broadcaster with the current configuration.
If no configuration is set, the broadcaster will not start.
To view the current configuration, use the ``show`` command.
```

The optional argument sets the index of the BIG to start.

Usage:
  .. parsed-literal::
     :class: highlight

     bct start [BIG_index]

> For example:

Reviewer 3: ```suggestion
Examples:
```

  .. parsed-literal::
     :class: highlight

     bct start

> This will start any BIG that is configured.

Reviewer 3: Please move all these lines (describing the examples) before the syntax of the actual example

> or

Reviewer 3: ```suggestion
```

  .. parsed-literal::
     :class: highlight

     bct start 0

This will start only BIG0.

stop - Stop broadcaster
=======================

> The command stops the broadcaster. The optional argument sets the index of the BIG to stop.

Reviewer 3: ```suggestion
The command stops the broadcaster.
The optional argument sets the index of the BIG to stop.
```

Please fix all the instances of sentences not starting on a new line.

  .. parsed-literal::
     :class: highlight

     bct location print

For example:
  .. parsed-literal::
     :class: highlight

     bct location fl 0 0 0

This will set the location for BIS 0 in subgroup 0 in BIG0 to Front Left.

broadcast_name - Set broadcast name
===================================

The command sets the broadcast name metadata for a BIG.

Usage:
  .. parsed-literal::
     :class: highlight

     bct broadcast_name <name> <BIG index>

For example:
  .. parsed-literal::
     :class: highlight

     bct broadcast_name Lecture 0

This will set the broadcast name for BIG0 to "Lecture".

For longer names with spaces, use " ".

For example:
  .. parsed-literal::
     :class: highlight

     bct broadcast_name "Lecture hall" 0

encrypt - Set broadcast code
============================

The command enables or disables encryption for a given BIG.

Usage:
  .. parsed-literal::
     :class: highlight

     bct encrypt <0/1> <BIG index> [broadcast_code]

The broadcast code is a 16-character string that is used to encrypt the broadcast, but shorter codes are also possible.

For example:
  .. parsed-literal::
     :class: highlight

>      bct encrypt 1 0 Auratest

Reviewer: So the first "1" is to enable ?  That is not fully clear.

Author: The documentation says to use either 0 or 1 as the first argument, and then I included an example for enabling and one example for disabling. How would you make it more clear?

This will enable encryption for BIG0 with the broadcast code "Auratest".

To disable encryption for BIG0 after setting it, use:

  .. parsed-literal::
     :class: highlight

     bct encrypt 0 0

usecase - Set use case
======================

The command sets a pre-defined use case.

Usage:
  .. parsed-literal::
     :class: highlight

     bct usecase <use_case>

> The use_case argument can either be an index or the name of a use case. The list all pre-defined use cases, type:

Reviewer: Please elaborate a bit on use cases.

phy - Set PHY
=============

The command sets the PHY for a BIG.

Usage:
  .. parsed-literal::
     :class: highlight

     bct phy <phy> <BIG index> <subgroup index>

The supported PHY values are:

  * 1 - 1M
  * 2 - 2M
  * 4 - Coded

For example:
  .. parsed-literal::
     :class: highlight

     bct phy 2 0 0

This will set the PHY for subgroup 0 in BIG0 to 2M.

framing - Set framing
=====================

The command sets the framing for a subgroup.

Usage:
  .. parsed-literal::
     :class: highlight

     bct framing <unframed/framed> <BIG index> <subgroup index>

For example:
  .. parsed-literal::
     :class: highlight

     bct framing unframed 0 0

This will set the framing for subgroup 0 in BIG0 to unframed.

rtn - Set number of re-transmits
================================

The command sets the number of re-transmits for a subgroup.

Usage:
  .. parsed-literal::
     :class: highlight

     bct rtn <number> <BIG index> <subgroup index>

For example:
  .. parsed-literal::
     :class: highlight

     bct rtn 2 0 0

This will set the number of re-transmits for subgroup 0 in BIG0 to 2.

sdu - Set SDU
=============

> The command sets the SDU size in octets for a subgroup.

Reviewer: ```suggestion
The command sets the SDU (Service Data Unit) size in octets for a subgroup.
```

Usage:
  .. parsed-literal::
     :class: highlight

     bct sdu <octets> <BIG index> <subgroup index>

For example:
  .. parsed-literal::
     :class: highlight

     bct sdu 60 0 0

This will set the SDU size for subgroup 0 in BIG0 to 60 octets.
Note that this will not change the bitrate, only the size of the SDU.

mtl - Set Max Transport latency
===============================

The command sets the Max Transport Latency for a subgroup.

Usage:
  .. parsed-literal::
     :class: highlight

     bct mtl <ms> <BIG index> <subgroup index>

For example:
  .. parsed-literal::
     :class: highlight

     bct mtl 10 0 0

This will set the Max Transport Latency for subgroup 0 in BIG0 to 10 ms.

frame_interval - Set frame interval
===================================

> The command sets the frame interval for a subgroup.

Reviewer 2: ```suggestion
The command sets the frame interval in microseconds for a subgroup.
```

Usage:
  .. parsed-literal::
     :class: highlight

     bct frame_interval <us> <BIG index> <subgroup index>

The supported values are:
  * 7500
  * 10000

For example:
  .. parsed-literal::
     :class: highlight

     bct frame_interval 10000 0 0

This will set the frame interval for subgroup 0 in BIG0 to 10 ms (10000 us).

pd - Set presentation delay
===========================

> The command sets the presentation delay for a subgroup.

Reviewer 2: ```suggestion
The command sets the presentation delay in microseconds for a subgroup.
```

Usage:
  .. parsed-literal::
     :class: highlight

     bct pd <us> <BIG index> <subgroup index>

For example:
  .. parsed-literal::
     :class: highlight

     bct pd 40000 0 0

This will set the presentation delay for subgroup 0 in BIG0 to 40 ms (40000 us).

.. _broadcast_configuration_tool_configuration:

Configuration
*************

> The application is pre-configured with a default configuration that is used when the application is started.

Reviewer 2: Is this true? When I run `bct start` without setting up any streams it doesn't look like it's broadcasting anything

Author: It is regarding the CONFIGs, I can make that more clear

For information about how to configure applications in the |NCS|, see :ref:`configure_application`.

.. _broadcast_configuration_tool_building:

Building and running
********************

> This application can be found under :file:`applications/audio/broadcast_config_tool` in the nRF Connect SDK folder structure.

Reviewer 2: Old path

> The sample is pre-configured with a default configuration in terms of memory allocated. This will limit the maximum numbers of BIGs, subgroups and BISes, but this can be changed in the prj.conf files. Even though the allocation allows for a certain amount of streams, there might not be air time for them all, so do your own calculations in terms of bitrates, re-transmits, etc.

Reviewer: ```suggestion
The sample is preset with a default configuration in terms of memory allocated. The default configuration is generous, and will allow for a large number of use cases. Defaults can be changed in the prj.conf files. There might not be air time for all streams if aggressive options are used. This depends on the large number of options such as bitrates, re-transmits, PHY etc.
```

> The sample is preset with a default configuration in terms of memory allocated. The default configuration is generous, and will allow for a large number of use cases. Defaults can be changed in the prj.conf files. There might not be air time for all streams if aggressive options are used. This depends on the large number of options such as bitrates, re-transmits, PHY etc.

Reviewer 3: ```suggestion
The sample is pre-configured with a generous default memory allocation, suitable for a wide range of use cases.
You can modify these default settings in the :file:`prj.conf` file.
Using aggressive configurations can reduce air time availability for all streams, depending on the combination of options selected (like high bitrates, increased re-transmits, specific PHY settings).
```
